library/alloc/src/slice/tests.rs

```diff
@@ -197,8 +197,7 @@ fn panic_safe() {
 
     let mut rng = test_rng();
 
-    // Miri is too slow (but still need to `chain` to make the types match)
-    let lens = if cfg!(miri) { (1..10).chain(0..0) } else { (1..20).chain(70..MAX_LEN) };
+    let lens = if cfg!(miri) { (1..10).chain(30..36) } else { (1..20).chain(70..MAX_LEN) };
```

Comment:
is there a reason why the comment was removed?

Reply:
It now does `.chain(30..36)`. I think there is value in testing an input length region that isn't just insertion sort. Arguably the stdlib sort tests are a little lacking, especially given the type specialization. It might make sense in the future to port parts or the whole sort-research-rs test suite into this project.

Reply:
ahh ok yeah reading the whole line I see what you mean, I jumped because of the first part "Miri is too slow" which still seems to be relevant?

Reply:
I'd say the code is self explanatory, there are many places in the stdlib tests that treat miri special, I'd say the issue is understood.

Reply:
Have you tried how long this takes in Miri? The is at least  quadratic  in the length, and may easily take many minutes in Miri.

Reply:
Good point. IIRC that change was based on our own test suite going up to input length [400](https://github.com/Voultapher/sort-research-rs/blob/438fad5d0495f65d4b72aa87f0b62fc96611dff3/sort_test_tools/src/tests.rs#L16C32-L17C5) for miri runs, that takes a couple seconds though.

How can we check the test execution time?

Reply:
Also I think as a stopgap, it's important that miri touches parts other than the insertion sort. That's like testing 20 LoC out of 1k, it's hardly interesting in terms of finding bugs. The threshold is `len <= 20`.

Reply:
Then maybe we should test length 21 (and only exactly that) in Miri. Or have a test that is less exhaustive inside the loop.

You can run this in Miri with `./x.py miri library/alloc -- panic_safe`.

Reply:
FWIW `test_sort` is entirely disabled in Miri. Maybe it's worth finding a configuration for that that's useful and doesn't take too long.

Reply:
Test run-time results on my Zen 3 machine:

- 1..10: 13s
- 1..10 + 21..22: 25s
- 1..10 + 30..36: 240s

> Then maybe we should test length 21 (and only exactly that) in Miri. Or have a test that is less exhaustive inside the loop.

Well then we'd still just test the small-sort, albeit more sophisticated it doesn't touch partitioning and the main quicksort loop. Given the test run-time results above and the fact that it would at best only test a limited portion of the code, I think it's best to just go back to only testing 1..10 even if that doesn't tell us much about the code. A more complete test solution exists, and as discussed before in this PR the medium-term plan is to port that test suite to the rust project to get better test coverage, including more miri coverage among other things.

Reply:
240s for a single test is definitely too long. I leave it up to you whether you want to add `21`. There's probably a way to get some panic-safety test coverage without being quote so exhaustive, like reducing some of the inner loops, but that's orthogonal to this PR.

Reply:
In my opinion adding `21` is a half measure with dubious benefits. I'd prefer to focus on getting this merged and then working on porting our more thorough test suite that includes various panic safety scenarios, and tests them different kinds of types to cover the specialization.